### PR TITLE
Fix/peer to peer with basic auth

### PR DESCRIPTION
**Pull request by euniceatcouchbase**
The change is to fix the test_peer_to_peer_with_basic_auth_incorrect_pass test case failure.
- corrected the java definition of ListenerAuthentication delegate
- fixed iOS code for peer to peer server start function
- deleted PasswordAuthentication class in java which is not needed

On python test framework code, I added exception message to lower case to assert unauthorized exception.

### CODE REVIEW

**Review by sridevi-15 — APPROVED** [review]

**Review by mgcouchbase — APPROVED** [review]

File: CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS/Server/PeerToPeerRequestHandler.swift [hunk 1]
```diff
@@ -97,9 +98,9 @@ public class PeerToPeerRequestHandler {
                 print("========== Setting CreateIdentity =========")
             }
 
-             if let auth: ListenerAuthenticator = args.get(name: "basic_auth") {
-                config.authenticator = auth
-             }
+            if auth != nil {
+               config.authenticator = auth
+            }
```
Comment (jayahariv) [inline CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS/Server/PeerToPeerRequestHandler.swift:103]: `args.get(name: "basic_auth")!` this is forced unwrapping variable above, returned value shouldn't be a `nil` right? 

what was wrong with the previous `if let` condition?

Comment (euniceatcouchbase) [reply CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS/Server/PeerToPeerRequestHandler.swift:103]: the previous if let didn't set the authenticator, I am lack of experience with swift, not sure why it caused authenticator a null value (shown in log)

Comment (euniceatcouchbase) [reply CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS/Server/PeerToPeerRequestHandler.swift:103]: basic_auth may or may not be set, if not basic auth method is used (instead, may be tls or something else), then config.authenticator will not be set

Comment (jayahariv) [reply CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS/Server/PeerToPeerRequestHandler.swift:103]: if the `basic_auth` is not set, is it going to be `(null)` or `nil`? both have different meaning.  From the above I assume, its '(null)', then it's okay to force unwrap. 

Still, I am not sure, how (i)assigning to a variable and then check for nil, vs (ii)if let declaration,  is different. For now, we can keep this way, but I would recommend we could look at this together may be after Lithium?

Comment (euniceatcouchbase) [reply CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS/Server/PeerToPeerRequestHandler.swift:103]: let me schedule a quick chat in zoom, I have another question to ask you.

File: CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS/Server/PeerToPeerRequestHandler.swift [hunk 2]
```diff
@@ -71,6 +71,7 @@ public class PeerToPeerRequestHandler {
             let tlsAuthType: String = args.get(name:"tls_auth_type")!
             let tlsAuthenticator: Bool? = args.get(name: "tls_authenticator")!
             let enable_delta_sync: Bool? = args.get(name: "enable_delta_sync")!
+            let auth: ListenerAuthenticator? = args.get(name: "basic_auth")!
```
[the remark this answers is not in the document]
Comment (euniceatcouchbase) [reply CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS/Server/PeerToPeerRequestHandler.swift]: I removed this line, had some code that Jay suggested.

**Review by pasin — APPROVED** [review]

**Review by jayahariv — APPROVED** [review]

**Pull request merged**